File: 452.cpp
#include <algorithm>

struct Solution {
    int findMinArrowShots(vector<vector<int>>& points) {
        sort(points.begin(), points.end(),
             [](const vector<int> &a, const vector<int> &b){return a[1] < b[1];});
        int res = 1, pos = points[0][1];
        for (int i = 1; i < points.size(); ++i){
            if (pos < points[i][0]){
                pos = points[i][1];
                ++res;
            }
        }
        return res;
    }
};

File: sol/452/452.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort(key=lambda itv: itv[1])
        res = 0
        pos = points[0][0] - 1
        for itv in points:
            if pos < itv[0]:
                pos = itv[1]
                res += 1
        return res
